src/pages/app/entry.js

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import firebase from 'gatsby-plugin-firebase';
+import { navigate } from 'gatsby';
+import { motion } from 'framer-motion';
+import moment from 'moment';
+import { Layout, Row, Col, Divider, Steps, Button, message } from 'antd';
+import debug from 'debug';
+import Delivery from '../../components/delivery';
+import { GraphQLNonNull } from 'graphql';
+
+function EntryPage({ location }) {
+	// Only allow entry page if there is a key
+	if (!location.search) return navigate('/');
+	const params = new URLSearchParams(location.search);
+	const key = params.get('key');
+
+	// Only allow entry page if there is a key
+	if (!key || key === '') return navigate('/');
```

Comment:
Try adding `const [shouldLeave, setShouldLeave] = useState(false);`

Then, on lines 14 and 19, change the return navigates to `setShouldLeave(true);`

THEN, after line 19, add this:

```javascript
useEffect(() => {
    if (window && shouldLeave) navigate('/');
}, [shouldLeave]);
```